### PR TITLE
Add a development standards section to README

### PR DESCRIPTION
**Pull request by bion**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,3 +11,29 @@ After this, you can access the server at localhost:9000.
...
+
+Prefer using immutable collection types provided by [Guava](https://github.com/google/guava) ([API docs](https://guava.dev/releases/snapshot/api/docs/)) over the Java standard library's mutable collections unless impractical. Include a comment justifying the use of a mutable collection if you use one.
+
+#### Separation of concerns
+
+In lieu of a microservice architecture, this project necessitates special care in ensuring separation of concerns. While Play provides some structure in this regard, it should be viewed as a starting point.
+
+Code in *Play controllers* should be limited to brokering interaction between the server's business logic systems and HTTP. Code in controller's should never directly implement business logic concerns and should instead delegate to classes specific to those purposes. One way to help think about this when writing controller code: if you're writing an HTTP handler that responds with HTML, factor out business logic classes so that implementing another handler that performs the same logic but responds with JSON benefits from high code re-use.
```
Comment (ndmckinley) [inline README.md:31]: I'd consider a link to your favorite primer on MVC - I'm a bit out of date on that, honestly.

File: README.md [hunk 2]
```diff
@@ -10,3 +11,29 @@ After this, you can access the server at localhost:9000.
...
+
+Code in *ebean models* should be limited to brokering interaction between the server's business logic and the database. Code in models should never directly implement business logic concerns.
+
+### Testing
+
+We aim for complete unit test coverage of all execution paths in the system. If you submit code that is infeasible or impractical to get full test coverage for, consider refactoring. If you would like to make an exception, include a clear explanation for why in your PR description.
```
Comment (ndmckinley) [inline README.md:37]: What's your position on integration / load testing?  My initial take is "let's use mocks and fakes as much as we can and implement the real integration stuff once we have a target platform", is that similar to yours?

**Review by ndmckinley — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (bion) [reply README.md]: Hmm I'll have to look around for one. If I can't find something that fits the bill I can give a quick talk like you did for Docker.

File: README.md (hunk 2 above)
Comment (bion) [reply README.md:37]: My guess is we won't have time or scope to do much in the way of load testing, and load is unlikely to ever be high enough to be much of a concern for scaling.

I'm pretty leery of relying too much on mocks and stubs for testing anything internal to the server, see go/tott/248 and go/tott/289

But it sounds like you're thinking in terms of testing things with dependencies on the other side of a network boundary, in which case yeah I think mocking/faking is our path forward.

File: README.md [hunk 3]
```diff
@@ -10,3 +11,29 @@ After this, you can access the server at localhost:9000.
...
+
+### Java code
+
+Java code should conform to the Google Java [styleguide](https://google.github.io/styleguide/javaguide.html). The project makes use of a linter and autoformatter for Java to help with this.
+
+Prefer using immutable collection types provided by [Guava](https://github.com/google/guava) ([API docs](https://guava.dev/releases/snapshot/api/docs/)) over the Java standard library's mutable collections unless impractical. Include a comment justifying the use of a mutable collection if you use one.
+
+#### Separation of concerns
+
+In lieu of a microservice architecture, this project necessitates special care in ensuring separation of concerns. While Play provides some structure in this regard, it should be viewed as a starting point.
```
[the remark this answers is not in the document]
Comment (bion) [reply README.md:29]: Yep, same idea. Thanks for the comment, I added a wikipedia link to help clarify.

File: README.md (hunk 2 above)
Comment (ndmckinley) [reply README.md:39]: Yep, thinking of databases and similar.  :)

**Review by hmhsu — APPROVED** [review]
Looks great! Thanks for adding this.

**Pull request merged**